js/web/lib/wasm/jsep/util.ts

```diff
@@ -485,6 +485,60 @@ export class GemmUtil {
   }
 }
 
+export const MIN_CLIP: Map<string, number> =
+    new Map([['float32', -3.4028234663852886e+38], ['float16', -65504.0], ['int32', -2147483648]]);
+
+export const MAX_CLIP: Map<string, number> =
+    new Map([['float32', 3.4028234663852886e+38], ['float16', 65504.0], ['int32', 2147483648]]);
+
+// Refer http://www.fox-toolkit.org/ftp/fasthalffloatconversion.pdf
+export const convertFloat16ToFloat32 = (float16Values: Uint16Array): Float32Array => {
```

Comment:
My suggestion is to avoid converting to/from float16 in JavaScript.

There are 2 types of data that we are working on:
- attribute
- tensor

For attributes, there is no float16 type. So all float attributes are float32. If we want to upload attributes as uniform, we should not convert them to f16 in CPU; instead, we can convert the uploaded f32 value to f16 in WGSL.

For tensors, it is possible to have float16 CPU data. If we want to use float16 tensor data as uniform, the best way is to use `getUint16Array()` and upload the data as-is. You may see strange numbers (eg. you may see 15565 while the actual value should be 1.2). But this is fine, because no calculation is performed in CPU. After uploading to GPU, you can use `f16` in WGSL as type to access the float16 data.

Specifically, for `Clip`, considering the operator changed to use inputs instead of attributes in v11, which is a quite old version, it is safe to assume most models uses inputs instead of attributes especially when using f16. If input/output is f16 but min/max is from attribute, just throw not supported (may need to check the actual model to determine whether my assumption is right ) hope this can reduce the complexity of the f16 support implementation.

Reply:
I agree that we should avoid converting to/from float16 in JavaScript.

>If input/output is f16 but min/max is from attribute, just throw not supported 

Maybe just use uniforms for attributes (< v11). I think it won't bring too much complexity.